packages/terra-pills/src/private/_Pill.jsx

```diff
@@ -0,0 +1,301 @@
+    pillInteractionHint = `, ${intl.formatMessage({ id: 'Terra.pills.hint.removable' })}`;
+  }
+
+  const pillClassNames = classNames(
+    cx([
+      'pill-label',
+      { 'is-focusable': pillInteraction.isSelectable || pillInteraction.isRemovable },
+      { 'is-removable': pillInteraction.isRemovable && !pillInteraction.isSelectableAndRemovable },
+      { 'is-selectable-and-removable': pillInteraction.isSelectableAndRemovable },
+      theme.className,
+    ]),
+    customProps.className,
+  );
+
+  const renderTruncatedLabelPopup = () => (
+    <Popup
+      isOpen={open}
+      isArrowDisplayed
+      targetRef={getPillRef}
+      onRequestClose={handleOnRequestClose}
+      contentHeight="auto"
+      contentWidth="auto"
+    >
+      <div className={cx(['popup-content-pill-label', theme.className])}>
+        {labelCategory
+          ? <span className={cx(['popup-content-pill-category', theme.className])}>{`${labelCategory}: `}</span>
+          : undefined}
+        <span>{label}</span>
+      </div>
+    </Popup>
+  );
+
+  const renderRemoveButton = () => (
+    <button
+      {...removeButtonProps}
+      className={cx('pill-remove-button')}
+      tabIndex="-1"
+      type="button"
+      aria-label={intl.formatMessage({ id: 'Terra.pills.label.delete' }, { pillLabelName: label })}
+    >
+      <span className={cx('clear-icon')} />
+    </button>
+  );
+
+  const renderSelectablePill = () => {
+    const categoryLabel = labelCategory || '';
+    const visuallyHiddenTextHint = ` ${categoryLabel} ${label} ${pillInteractionHint}`;
+
+    return (
+      <>
+        <div
+          {...customProps}
+          {...pillProps}
+          {...pillButtonProps}
+          aria-haspopup={pillInteraction.isSelectable ? 'dialog' : undefined}
+          id={id}
+          className={cx(['pill', 'is-focusable'])}
+          ref={pillRef}
+          role="listitem"
+          aria-label={visuallyHiddenTextHint}
+          data-terra-pills-show-focus-styles
+          data-terra-pill
+        >
+          <span className={pillClassNames}>
+            {labelCategory
+              ? <span className={cx('pill-category')}>{`${labelCategory}: `}</span>
+              : undefined}
+            <span>{label}</span>
+          </span>
+          {pillInteraction.isRemovable && renderRemoveButton()}
+        </div>
+        {isTruncated && renderTruncatedLabelPopup()}
+      </>
+    );
+  };
+
+  const renderBasicPill = () => {
+    const categoryLabel = labelCategory || '';
+    const visuallyHiddenTextHint = ` ${categoryLabel} ${label} ${pillInteractionHint}`;
+    return (
+      <>
+        <div
+          {...customProps}
+          {...pillProps}
+          id={id}
+          className={cx(['pill', 'is-focusable'])}
+          ref={pillRef}
+          role="listitem"
+          aria-label={visuallyHiddenTextHint}
+          data-terra-pills-show-focus-styles
+          data-terra-pill
+        >
+          <label className={pillClassNames}>
```

Comment:
Is there a reason why we're using a `label` element here but a `span` element in renderSelectablePill?

Reply:
It was changed to 'label' to fix few accessibility issues during development stages. Somehow was left forgotten.

Reverted to `span` here cfb6aad5b38a9957f1449aa2f75bbf5d001c3c96